### DISCUSSION TITLE
Unable to serialize AdminStatusEnum after implemented doctrine Enum

### DISCUSSION THREAD

**Discussion by storexstore** [dotkernel/api#420] [General]
{"error":{"messages":["Encountered object of type \u0022Api\\Admin\\Enum\\AdminStatusEnum\u0022 when serializing Mezzio\\Hal\\HalResource instance; unable to serialize"]}}

**Comment by alexmerlin** [comment]
Hey @hiusing

Could you provide more details on how you encountered this issue?
Did it happen on calling a specific endpoint?
If not, please send a code snippet on how to reproduce the issue.

**Comment by storexstore** [comment]
Hi @alexmerlin 

https://www.dotkernel.com/dotkernel/doctrine-enum-implementation-in-dotkernel/

The latest version of dotkernel api implement the Doctrine EnumType columns in the Admin and User.  Mezzio\Hal\HalResource doesn't know how to serialize the AdminStatusEnumType.

When you composer serve and hit the endpoint at http://localhost:8080/admin to list all admin.  You encounter the following error message at the brower:

  | {"error":{"messages":["Encountered object of type \u0022Api\\Admin\\Enum\\AdminStatusEnum\u0022 when serializing Mezzio\\Hal\\HalResource instance; unable to serialize"]}}

**Comment by alexmerlin** [comment]
@hiusing 

Did you modify anything in your copy of Dotkernel API before running it and encountering this error?
Does this behaviour persist with other endpoints as well? i.e.: `/admin/role`, `/users`, `/admin/my-account`

### Off-topic:
We strongly recommend using virtualhosts for running your applications.
They provide a low-effort solution for running applications without the need for manually starting them every time.
See our documentation on [setting up WSL2](https://docs.dotkernel.org/development/v1/introduction/) and [creating virtualhosts in a WSL2 Linux distro](https://docs.dotkernel.org/development/v1/virtualhosts/overview/) if you're on WIndows.
If you are on a different operating system, you can find tutorials on the internet.

---

### On-topic:
I just tested step-by-step using your method and could not reproduce the issue you are describing.

---

### Recommendations:
* make sure you are [running the application in development mode](https://docs.dotkernel.org/api-documentation/v5/installation/composer/#development-mode) so that config modifications are immediately available, without the need for clearing the cache first
* make sure that the [custom enum types](https://github.com/dotkernel/api/blob/5.0/config/autoload/doctrine.global.php#L36-L38) are registered in your `config/autoload/doctrine.global.php` file
* navigate to the root of your project and run the tests by executing the below command:
(If present in the `log` directory, delete the current day's log file `error-log-2024-11-XX.log` else the tests might fail due to file permission issues)

```shell
composer test
```

If the tests succeed, it means that the serialization works.

## Conclusion
Let us know your server specifications so we can narrow down possible causes of the issue you're describing:
  * operating system
  * PHP version
  * database type and client/server version
  * your Dotkernel API version (see the most recent release in your `CHANGELOG.md` file)
